libs/core/src/lib/split-button/split-button.component.ts

```diff
@@ -79,9 +92,71 @@ export class SplitButtonComponent {
     @ContentChild(MenuComponent)
     menu: MenuComponent;
 
+    /** @hidden */
+    @ViewChild('mainActionButton', { read: ElementRef })
+    mainAction: ElementRef;
+
+    /** @hidden */
+    mainButtonWidth: number;
+
+    /** @hidden */
+    private _menuItemSubscriptions = new Subscription();
+
+    /** @hidden */
+    constructor(private _cdRef: ChangeDetectorRef, private _elRef: ElementRef) {}
+
     /** @hidden Emits event when main button is clicked */
     onMainButtonClick(event: MouseEvent): void {
         this.primaryButtonClicked.emit(event);
         event.stopPropagation();
     }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this._setupMenuItemSubscriptions();
+
+        if (!this.mainActionTitle && !this.selected) {
+            this.selectMenuItem(this.menu.menuItems.first);
+        } else if (this.selected) {
+            this.selectMenuItem(this.selected);
+        }
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        if (this.fixedWidth) {
+            this.mainButtonWidth = parseInt(this.mainAction.nativeElement.offsetWidth, 10);
+        }
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._menuItemSubscriptions.unsubscribe();
+    }
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes && changes.selected) {
+            this.selectMenuItem(this.selected);
+        }
+    }
+
+    /** Function called to select a menu item for the split button */
+    selectMenuItem(menuItem: MenuItemComponent): void {
+        if (menuItem) {
+            menuItem.setSelected(true);
+        }
+    }
+
+    /** @hidden */
+    private _setupMenuItemSubscriptions(): void {
+        this.menu.menuItems.map((menuItem: MenuItemComponent) => {
+            this._menuItemSubscriptions.add(
+                menuItem.onSelect.subscribe(() => {
+                    this.mainActionTitle = menuItem.menuItemTitle.title;
+                    this._cdRef.detectChanges();
+                })
+            );
+        });
+    }
```

Comment:
You would need to listen for changes
![dynamic split button](https://user-images.githubusercontent.com/26483208/92706372-bf451700-f354-11ea-8785-dca60ca74d3a.gif)
Because dynamically addded options are not supported